crates/bevy_reflect/bevy_reflect_derive/src/container_attributes.rs

```diff
@@ -0,0 +1,155 @@
+use proc_macro2::{Ident, Span};
+use quote::quote;
+use syn::parse::{Parse, ParseStream};
+use syn::punctuated::Punctuated;
+use syn::token::Comma;
+use syn::{Meta, NestedMeta, Path};
+
+#[derive(Clone)]
+pub enum TraitImpl {
```

Comment:
Doc strings please; this is pub.

Reply:
Sorry, my visibilities are a bit off because of I prefer marking methods `pub` instead of `pub(crate)` which requires its params be `pub` as well. This isn't actually user facing, but I'll try  to fix up the visibilities to reflect this better.

Reply:
Yes please; that will help both reviewers and doc readers.